tests/test_plugin_coinbase_pro.py

```diff
@@ -0,0 +1,118 @@
+from typing import List, Dict, Any
```

Comment:
This is really cool: I hadn't used pytest-mock before. A couple of minor nits:
- can you add a copyright and license header similar to the one at the beginning of all other source files (e.g. tests/test_ods_output_diff.py)
- can you use Python unit tests (they are used both in DaLI and RP2). You don't have to do it, if you don't have time: I can migrate it later.

Reply:
For copyright license header, i recommend using tools like rat to enforce it during code review like other ASF projects if you feel strongly about it.

I changed the test function to a class to better align with the structure you have with tests using uniitest module. But honestly, I recommend stop using the unittest module going forward because it doesn't add much value. Pytest is designed to work with functions and fixtures. I tend to find tests declared in functions easier to read and reason because code is less nested and dependencies are clearly declared in the function signatures. With a class style enforced by the unittest module, it's never clear how much states are shared between tests without scrolling back to the class init method to see what class attributes are defined. I have switched to pure function tests in all my python projects for many years and it scales very well for small to very large projects.

The biggest problem with the unnittest module is it doesn't play well with pytest's fixture feature, that's why it's recommended to not use it for pytest tests. See the first note at the bottom of https://pytest.org/en/latest/how-to/unittest.html. The unittest support was added to pytest only for backwards compatibility reasons.

Reply:
Thanks for the suggestions. I will definitely look into rat. As for unit tests, you make a compelling case: going forward, let's move away from unit test, or maybe allow both, so that it's up to the test creator to decide what to use.

Reply:
yeah, it's very typical to allow both as well, especially for projects with longer histories :)